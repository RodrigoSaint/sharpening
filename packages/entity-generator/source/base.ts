export type FieldType = "string" | "int" | "decimal" | "date";

export interface Field {
  name: string;
  isRequired: boolean;
  type: FieldType;
}

export type EntityAction =
  | "pagination"
  | "list"
  | "detail"
  | "create"
  | "update"
  | "delete";

export interface Entity {
  name: string;
  fieldCollection: Field[];
  actionCollection?: EntityAction[];
  formFieldCollection?: string[];
  displayFieldCollection?: string[];
}
