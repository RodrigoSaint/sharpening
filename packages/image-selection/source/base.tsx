import React, { useState } from "react";
import Modal from "@rodrigosaint/modal";
import ManagedImage from "@rodrigosaint/managed-image";

import ImageSelectionModalContent, {
  ImageSelectionModalContentProps,
} from "./content";

export interface ImageSelectionProps<T>
  extends Omit<ImageSelectionModalContentProps<T>, "close"> {
  renderWithoutImage: () => React.ReactNode;
  getImageUrl: (item: T) => string;
  header: string | React.ReactNode;
  initialValue?: T;
}

export default function ImageSelection<T>(props: ImageSelectionProps<T>) {
  const [src, setSrc] = useState<string>(
    props.initialValue ? props.getImageUrl(props.initialValue) : undefined
  );

  const onSelected = React.useCallback(
    (item: T) => {
      setSrc(props.getImageUrl(item));
      props.onSelected(item);
    },
    [props.onSelected, props.getImageUrl]
  );

  return (
    <>
      <Modal
        renderTrigger={() => (
          <ManagedImage
            src={src}
            renderWithoutImage={props.renderWithoutImage}
            renderOnError={props.renderWithoutImage}
          />
        )}
        renderContent={(close) => (
          <ImageSelectionModalContent
            {...props}
            onSelected={onSelected}
            close={close}
          />
        )}
        header={props.header}
      />
    </>
  );
}
