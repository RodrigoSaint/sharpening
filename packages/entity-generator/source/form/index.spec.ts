import { Entity } from "../base";
import { getForm, getCreateHook, getUpdateHook } from ".";

const entity: Entity = {
  name: "User",
  fieldCollection: [
    {
      name: "name",
      isRequired: true,
      type: "string",
    },
    {
      name: "email",
      isRequired: false,
      type: "string",
    },
    {
      name: "age",
      isRequired: false,
      type: "int",
    },
  ],
  formFieldCollection: ["name", "email"],
};

describe("Form", () => {
  it("renders form properly", () => {
    expect(getForm(entity)).toMatchSnapshot("getForm");
  });
  it("renders create hook properly", () => {
    expect(getCreateHook(entity)).toMatchSnapshot("getCreateHook");
  });
  it("renders create hook properly", () => {
    expect(getUpdateHook(entity)).toMatchSnapshot("getUpdateHook");
  });
});
