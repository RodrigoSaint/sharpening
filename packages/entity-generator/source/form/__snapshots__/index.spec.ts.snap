// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders create hook properly: getCreateHook 1`] = `
"import { User, userSchema } from \\"entity/user\\";
import { useSimpleForm } from \\"@rodrigosaint/form\\";
import { useMutation } from \\"@rodrigosaint/graphql\\";

const USER_CREATE_MUTATION = \`
  mutation ($user: UserInput!){
    userCreate(user: $user) {
      _id
    }
  }\`;

export default function useUserCreate() {
  const form = useSimpleForm < User > userSchema;
  const userCreate =
    useMutation < { user: User } > (\\"user\\", USER_CREATE_MUTATION);

  return {
    onSubmit: form.handleSubmit((user) => userCreate({ user })),
    form,
  };
}
"
`;

exports[`Form renders create hook properly: getUpdateHook 1`] = `
"import { User, userSchema } from \\"entity/user\\";
import { useSimpleForm } from \\"@rodrigosaint/form\\";
import { useMutation } from \\"@rodrigosaint/graphql\\";

const USER_UPDATE_MUTATION = \`
  mutation ($user: UserInput!){
    userUpdate(user: $user) {
      _id
    }
  }\`;

export default function useUserUpdate() {
  const form = useSimpleForm < User > userSchema;
  const userUpdate =
    useMutation < { user: User } > (\\"user\\", USER_UPDATE_MUTATION);

  return {
    onSubmit: form.handleSubmit((user) => userUpdate({ user })),
    form,
  };
}
"
`;

exports[`Form renders form properly: getForm 1`] = `
"import Form from \\"@rodrigosaint/form\\";
import Field from \\"@rodrigosaint/field\\";
import Button from \\"@rodrigosaint/button\\";
import { UseFormReturn } from \\"react-hook-form\\";

import { User } from \\"entity/user\\";

interface UserFormProps {
  form: UseFormReturn<User>;
  onSubmit: () => Promise<any>;
}

export default function UserForm(props: UserFormProps) {
  return (
    <Form {...props}>
      <Field name=\\"name\\" />
      <Field name=\\"email\\" />
      <Button type=\\"submit\\">submit</Button>
    </Form>
  );
}
"
`;
